drivers/ddr/imx/imx8m/helper.c

```diff
@@ -23,6 +23,11 @@ DECLARE_GLOBAL_DATA_PTR;
 #define DMEM_OFFSET_ADDR 0x00054000
 #define DDR_TRAIN_CODE_BASE_ADDR IP2APB_DDRPHY_IPS_BASE_ADDR(0)
 
+#ifdef CONFIG_SPL_WITH_DDR_FIRMWARE
+char __firmware_image_start[0] __attribute__((section(".__firmware_image_start")));
```

Comment:
is there an alignment requirement here?

Reply:
It does need to be aligned, but I've specified that in the linker script:
https://github.com/mike-scott/u-boot/blob/fio-v2020.04-imx8m-ddr-firmware/arch/arm/cpu/armv8/u-boot-spl.lds#L54